### PR TITLE
[Feature] Encryption Key rotation

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

File: pkg/apis/deployment/v1/plan.go [hunk 1]
```diff
@@ -79,6 +79,12 @@ const (
 	ActionTypeBackupRestore ActionType = "BackupRestore"
 	// ActionTypeBackupRestoreClean restore plan
 	ActionTypeBackupRestoreClean ActionType = "BackupRestoreClean"
+	// ActionTypeEncryptionKeyAdd add new encryption key to list
+	ActionTypeEncryptionKeyAdd ActionType = "EncryptionKeyAdd"
+	// ActionTypeEncryptionKeyAdd removes encryption key to list
```
Comment (Robert-Stam) [inline pkg/apis/deployment/v1/plan.go:84]: ActionTypeEncryptionKeyAdd  --> ActionTypeEncryptionKeyRemove

File: pkg/deployment/client/tls.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Adam Janikowski
+//
+
+package client
+
+type TLSKeyFile struct {
+	PrivateKeyChecksum string   `json:"privateKeySHA256,omitempty"`
```
Comment (Robert-Stam) [inline pkg/deployment/client/tls.go:26]: PrivateKeyChecksum --> PrivateKeyHash?

File: pkg/deployment/pod/encryption.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GroupEncryptionSupported(mode api.DeploymentMode, group api.ServerGroup) bool {
+	switch mode {
+	case api.DeploymentModeCluster:
+		switch group {
+		case api.ServerGroupDBServers:
+			fallthrough
+		case api.ServerGroupAgents:
```
Comment (Robert-Stam) [inline pkg/deployment/pod/encryption.go:44]: can be combined to `case api.ServerGroupDBServers, api.ServerGroupAgents:`

File: pkg/deployment/pod/encryption.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+	case api.DeploymentModeSingle:
+		return group == api.ServerGroupSingle
+
+	case api.DeploymentModeActiveFailover:
+		switch group {
+		case api.ServerGroupSingle:
+			fallthrough
+		case api.ServerGroupAgents:
+			return true
```
Comment (Robert-Stam) [inline pkg/deployment/pod/encryption.go:58]: combine?

File: pkg/deployment/pod/jwt.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+func JWT() Builder {
+	return jwt{}
+}
+
+type jwt struct{}
+
+func (j jwt) Args(i Input) k8sutil.OptionPairs {
+	panic("implement me")
```
Comment (Robert-Stam) [inline pkg/deployment/pod/jwt.go:45]: todo? (3 times)

File: pkg/apis/backup/v1/backup_spec.go [hunk 6]
```diff
@@ -26,6 +26,8 @@ type ArangoBackupSpec struct {
 	// Deployment
 	Deployment ArangoBackupSpecDeployment `json:"deployment,omitempty"`
 
+	EncryptionSecret *string `json:"encryptionSecret,omitempty"`
```
Comment (Robert-Stam) [inline pkg/apis/backup/v1/backup_spec.go:29]: EncryptionSecret  --> EncryptionSecretName ?

I am not sure how to rotate a secret, however if it's allowed to remove the secret after the rotate is done, this name is useless.
Shouldn't it contain a hash of the used key (when a backup is created). And this name as part of the restore only?

File: pkg/deployment/pod/jwt.go (hunk 5 above)
Comment (ajanikow) [reply pkg/deployment/pod/jwt.go:45]: Yes, this file will be removed

File: pkg/apis/backup/v1/backup_spec.go (hunk 6 above)
Comment (ajanikow) [reply pkg/apis/backup/v1/backup_spec.go:29]: This is related only to Backup - Key from this secret will be restored on all servers just for Restore Operation and removed after. But I can move it to Restore request in ArangoDeployment - good idea

To rotate EncryptionKey in ArangoDeployment - just change this secret or change secret name in ArangoDeployment.

**Pull request merged**